packages/server/src/adapters/ws.ts

```diff
@@ -38,10 +39,14 @@ export type CreateWSSContextFnOptions = NodeHTTPCreateContextFnOptions<
 export function applyWSSHandler<TRouter extends AnyRouter>(
   opts: WSSHandlerOptions<TRouter>,
 ) {
-  const { wss, createContext, router } = opts;
+  const { wss, createContext, router, prefix } = opts;
 
   const { transformer } = router._def._config;
   wss.on('connection', async (client, req) => {
+    if (prefix && req.url !== prefix) {
```

Comment:
should this be strict equality? can the url never have query params or similar?

Reply:
Indeed `req.url` contain the query params, we could do `req.url.startsWith` or `req.url.include`